src/base.ts

```diff
@@ -63,6 +63,12 @@ export class AztecWalletSdk {
       );
     });
 
+    this.#currentConnector.subscribe((connector) => {
+      if (connector) {
+        this.#reconnect();
+      }
+    });
+
```

Comment:
this should be put in the `#currectConnector` definition. Your code only subscribes to the current connector on the `AztecWalletSdk` instantiation. So, if user switches from Obsidion -> Azguard, it won't call `.reconnect()`. Could you try with this code and verify that it works as you expect?

```ts
 this.#currentConnector = reactive(($) => {
  const currentConnectorUuid = $(this.#currentConnectorUuid);
  const connector = $(this.#connectors).find(
    (a) => a.info.uuid === currentConnectorUuid,
  );

  // side effect. Is this correct?
  if (connector) {
    connector.reconnect();
  }

  return connector;
});
```

Reply:
@olehmisar `reactive.subscribe(fn)` triggers `fn` each time when the `reactive` value is updated. So, when a user switches to a different wallet by setting the `#currentConnectorUuid`, the `#currentConnector` is automatically updated, triggering the subscription, that triggers reconnection.

So, it actually should work.

But your code works too, so it's up to you.

Reply:
you are right. But I'd prefer my version as it does not have any perpetual subscriptions.